packages/providers/upload-aws-s3/lib/index.js

```diff
@@ -33,9 +41,13 @@ module.exports = {
               return reject(err);
             }
 
-            // set the bucket file url
-            file.url = data.Location;
-
+            // set the bucket file url. 
+            // If there is a Custom endpoint for data access set, replace the upload endpoint with the read enpoint URL
+            if(config.customReadEndpoint){
+              file.url = config.customReadEndpoint.replace(/\/$/g,'')+'/'+data.Key;
+            } else {
+              file.url = data.Location;
+            }
```

Comment:
This bit of code conflicts with an ongoing [PR ](https://github.com/strapi/strapi/pull/13040 to make this provider work with a CDN.

I was trying to think of alternatives but I can't think of any.

So I wanted to ask, would that change work for Cloudflare R2 & you? If yes, we can try to ping them to see how that is advancing.

Or we try to provide an alternative, maybe creating an external provider in the meantime.